lib/commands/create-project.ts

```diff
@@ -1,44 +1,207 @@
 
 export class CreateProjectCommand implements ICommand {
 	public enableHooks = false;
-	public allowedParameters: ICommandParameter[] = [this.$stringParameterBuilder.createMandatoryParameter("Project name cannot be empty.")];
+	public allowedParameters: ICommandParameter[] = [this.$stringParameter];
+	private static NgFlavor = "Angular";
+	private static VueFlavor = "Vue.js";
+	private static TsFlavor = "Plain TypeScript";
+	private static JsFlavor = "Plain JavaScript";
+	private static HelloWorldTemplateKey = "Hello World";
+	private static HelloWorldTemplateDescription = "A Hello World app";
+	private static DrawerTemplateKey = "SideDrawer";
+	private static DrawerTemplateDescription = "An app with pre-built pages that uses a drawer for navigation";
+	private static TabsTemplateKey = "Tabs";
+	private static TabsTemplateDescription = "An app with pre-built pages that uses tabs for navigation";
 
-	private createdProjecData: ICreateProjectData;
+	private createdProjectData: ICreateProjectData;
 
 	constructor(private $projectService: IProjectService,
 		private $logger: ILogger,
 		private $errors: IErrors,
 		private $options: IOptions,
-		private $stringParameterBuilder: IStringParameterBuilder) { }
+		private $prompter: IPrompter,
+		private $stringParameter: ICommandParameter) { }
 
 	public async execute(args: string[]): Promise<void> {
-		if ((this.$options.tsc || this.$options.ng) && this.$options.template) {
-			this.$errors.fail("You cannot use --ng or --tsc options together with --template.");
+		const interactiveAdverbs = ["First", "Next", "Finally"];
+		const getNextInteractiveAdverb = () => {
+			return interactiveAdverbs.shift() || "Next";
+		};
+
+		if ((this.$options.tsc || this.$options.ng || this.$options.vue || this.$options.js) && this.$options.template) {
+			this.$errors.fail("You cannot use a flavor option like --ng, --vue, --tsc and --js together with --template.");
 		}
 
+		let projectName = args[0];
 		let selectedTemplate: string;
-		if (this.$options.tsc) {
+		if (this.$options.js) {
+			selectedTemplate = constants.JAVASCRIPT_NAME;
+		} else if (this.$options.tsc) {
 			selectedTemplate = constants.TYPESCRIPT_NAME;
 		} else if (this.$options.ng) {
 			selectedTemplate = constants.ANGULAR_NAME;
+		} else if (this.$options.vue) {
+			selectedTemplate = constants.VUE_NAME;
 		} else {
 			selectedTemplate = this.$options.template;
 		}
 
-		this.createdProjecData = await this.$projectService.createProject({
-			projectName: args[0],
+		if ((!selectedTemplate || !projectName) && isInteractive()) {
+			this.printInteractiveCreationIntro();
+		}
+
+		if (!projectName && isInteractive()) {
+			projectName = await this.$prompter.getString(`${getNextInteractiveAdverb()}, what will be the name of your app?`, { allowEmpty: false });
+			this.$logger.info();
+		}
+
+		projectName = await this.$projectService.validateProjectName({ projectName: projectName, force: this.$options.force, pathToProject: this.$options.path });
```

Comment:
The validation must be before `printInteractiveCreationIntro` call, as if it fails the output is strange:
```
tns create ap

# Let’s create a NativeScript app!

Answer the following questions to help us build the right app for you. (Note: you
can skip this prompt next time using the --template option, or the --ng, --vue, --ts,
or --js flags.)

Path already exists and is not empty D:\Rosen\Work\nativescript-cli\scratch\ap
```